### ISSUE TITLE
Have rustdoc create links in compiled code examples

### ISSUE THREAD

**Issue by Manishearth** [rust-lang/rust#28954]
Currently we just generate HTML for the code examples by compiling markdown.

However, rustdoc could compile these examples with a plugin that emits resolved path information for each `ExprPath` and import being used. This part isn't too hard. Might actually be useful as a plugin to be run on independent rust programs.

Then, we need to tweak the markdown bits so that these paths (those with a known span, that is), get linked appropriately.

cc @alexcrichton @steveklabnik

**Comment by steveklabnik** [comment]
I believe we have a bug open for this already? It's been this way for so long, we have to have had...

**Comment by alexcrichton** [comment]
This is basically just DXR in rustdoc

**Comment by Manishearth** [comment]
Yeah, we realized that after discussing a bit.

**Comment by gambogi** [comment]
So is this issue still open then? Or is it a duplicate?

**Comment by Manishearth** [comment]
I don't think so, but DXR exists which gives a similar code search (just not in rustdoc examples). Not sure if this is worth pursuing.

**Comment by steveklabnik** [comment]
Triage: no change
